zellij-server/src/tab/mod.rs

```diff
@@ -2196,212 +2196,114 @@ impl Tab {
                 .unwrap();
         }
     }
-    pub fn scroll_active_terminal_up(&mut self, client_id: ClientId) {
+
+    fn operate_active_terminal_scroll<ClosureT>(
```

Comment:
Hey, this is great! As long as we're here, how about we do something like:

```rust
pub fn scroll_active_terminal_up(&mut self, client_id: ClientId) {
    // ...
    ScrollActiveTerminal::new(/* .. */).up(1)
}
```

Reply:
Or do you think that'll be too involved? I just feel it'll save us passing closures around and mucking about with generics.

Reply:
Hi @imsnif,

Thank you for your suggestion!
I agree we want to avoid closures and generics as possible. However, your suggestion initializes a struct as a closure initializes a struct as described [here](https://rustc-dev-guide.rust-lang.org/closure.html). So I'm thinking there may not be that much difference in terms of memory usage (but the code looks beautiful!).

I would like to additionally investigate if this code can be improved without closures and generics in three aspects: memory usage, speed, and code cleanliness :)

Reply:
Sure - as you wish. Just bear in mind that I don't think performance or resource utilization are super important in this flow. When scrolling the terminal we're bound by STDIN which is considerably slower.

My suggestion was mostly because I thought it would read better.

Reply:
Oh, that's certainly right... I'll follow your suggestion! Thank you!